test/subtype.jl

```diff
@@ -826,3 +826,28 @@ test_old()
 test_intersection()
 test_properties()
 test_intersection_properties()
+
+# Issue #12580
+abstract AbstractMyType12580{T}
+immutable MyType12580{T}<:AbstractMyType12580{T} end
+tpara{A<:AbstractMyType12580}(::Type{A}) = tpara(supertype(A))
+tpara{I}(::Type{AbstractMyType12580{I}}) = I
+@test tpara(MyType12580{true})
+
+# Issue #18348
+f18348{T<:Any}(::Type{T}, x) = 1
+f18348{T<:Any}(::Type{T}, x::T) = 2
+@test length(methods(f18348, Tuple{Type{Any},Any})) == 1
+
+# Issue #13165
+@test Symmetric{Float64,Matrix{Float64}} <: LinAlg.RealHermSymComplexHerm
+@test Hermitian{Float64,Matrix{Float64}} <: LinAlg.RealHermSymComplexHerm
+@test Hermitian{Complex{Float64},Matrix{Complex{Float64}}} <: LinAlg.RealHermSymComplexHerm
+
+# Issue #11803
+@test TypeVar(:T, Int) <: Int
```

Comment:
I don't like this test. Manually constructing a TypeVar is not really a supposed operation. I wonder if it would be fair to say that:
`(Tuple{T} where T<:Union{Int,Float64}) <: Tuple{Union{Int64,Float64}}` tests the same thing. @JeffBezanson?